packages/dataviews/src/dataviews.tsx

```diff
@@ -108,8 +119,9 @@ export default function DataViews< Item extends AnyItem >( {
 					newSelection.includes( getItemId( item ) )
 				)
 			);
+			lastUpdatedSelectionBecauseOfItemRemoval.current = selection;
 		}
-	}, [ selection, data, getItemId, onSelectionChange ] );
+	}, [ selection, data, getItemId, onSelectionChange, setSelection ] );
```

Comment:
I find this logic really hard to follow. To me it sounds like this should only be setting the initial selection on mount? Am I missing something?

Reply:
Hi @ellatrix, it is not only on mout that this effect is needed. This effect removes an item from the selection when an item is no longer in the data set. That may happen because the item was trashed or because the views/filters were changed etc.

Reply:
It sounds like the implementor of DataViews is responsible for that?

Reply:
If an something is selected that no longer exists, why not just ignore it?

Reply:
> It sounds like the implementor of DataViews is responsible for that?

Until very recently the selection was managed by Dataviews and the implementor had no ability to change it. I guess that was the reason why the effect was added in dataviews itself. That and to avoid all the implementors needing a have an effect like this.

> If something is selected that no longer exists, why not just ignore it?

That causes multiple UI issues where we show toolbars when items are selected and show the number of selected items. We would be showing information to the user that does not match what the user sees as selected.

Reply:
> That causes multiple UI issues where we show toolbars when items are selected and show the number of selected items. We would be showing information to the user that does not match what the user sees as selected.



Which toolbars? Inside or outside of DataViews?